### ISSUE TITLE
technical investigation for hyphenated diagnostic codes

### ISSUE THREAD

**Issue by amylai-va** [department-of-veterans-affairs/abd-vro#2686]
If we know whether a Veteran's disability is rated using a hyphenated diagnostic code (DC) and what those DCs are, then we would be more confident about what the maximum rating is for the disability. 

## Acceptance Criteria
- Investigate whether we could get both DCs from the BIP Claims API
- Investigate whether we could get both DCs from BGS
- Document findings & recommended approach

## Notes about this work
- In vets-api, rated disabilities data comes from EVSS or the Veteran Verification API. Both get their information from BGS (Benefits Gateway Services). Currently, only one DC is available.
- Field names from PA&I: `HYPNTD_DGNSTC_TYPE_CD` and `DGNSTC_TYPE_CD`

**Comment by nanotone** [comment]
tl;dr **It is possible to fetch hyphenated diagnostic from within vets-api.** 🥳 

Investigation breadcrumbs:
- It seems that the Veteran Verification v2 API [doesn't return hyphenated codes](https://github.com/department-of-veterans-affairs/abd-vro/issues/2500#issuecomment-1939648236), but when I asked the maintainers whether it would be possible to add, they mentioned that VV2 gets its [data from BGS](https://lighthouseva.slack.com/archives/CDW40SBTL/p1708535641944579?thread_ts=1707842143.983309&cid=CDW40SBTL).
- The field names in the PA&I dataset are `HYPNTD_DGNSTC_TYPE_CD` and `DGNSTC_TYPE_CD`. This style of vowel-less shortening is very characteristic of BGS/CorpDB, and it would be very unusual for only the primary DC to be available in BGS but not the hyphenated DC.
- For rated disabilities data in vets-api, VV2 is the planned replacement for EVSS. EVSS also uses BGS, and although it doesn't return hyphenations either, I learned from an EVSS dev that it calls the BGS [RatingInformationService with this SOAP payload](https://dsva.slack.com/archives/C02CQP3RFFX/p1697744153948009?thread_ts=1697050321.703279&cid=C02CQP3RFFX).
- vets-api's BGS library [does support RatingInformationService](https://github.com/department-of-veterans-affairs/bgs-ext/blob/master/lib/bgs/services/rating_profile.rb) but not this particular call -- and it doesn't seem like vets-api uses this service at all. Nonetheless, the library enables us to make such a call from vets-api's Ruby console.

Using our [known example](https://github.com/department-of-veterans-affairs/abd-vro/issues/2500) of a Veteran with a rated disability that has a hyphenated DC:

```ruby
> fsub = Form526Submission.find(1662508)
> icn = fsub.user_account.icn
> key = fsub.form['form526']['form526']['veteran']['emailAddress'].first(BGS::Constants::EXTERNAL_KEY_MAX_LENGTH)
> service = BGS::Services.new(external_uid: icn, external_key: key).rating_profile
> resp = service.send(:request, :read_current_disabilities, fileNbr: fsub.birls_id, latestEvalOnly: true)
> disabilities = resp.body[:read_current_disabilities_response][:disabilities]
```

Looking through the rated disabilities, I found the "tinnitus with insomnia" that we were interested in:
```ruby
> disabilities[3][:disability_evaluations]
=>
{:call_id=>"17",
 :name=>"DisabilityEvaluation",
 :begin_dt=>Tue, 07 Feb 2023 00:00:00 -0600,
 :dgnstc_tc=>"6260",
 :dgnstc_tn=>"Tinnitus",
 :dgnstc_txt=>"tinnitus with insomnia",
 :dgnstc_txt_clob=>"tinnitus with insomnia",
 :dis_dt=>Fri, 21 Jul 2023 16:41:24 -0500,
 :dis_sn=>"89334084",
 :hyp_dgnstc_tc=>"9423",
 :prcnt_no=>"30",
 :prfl_dt=>Fri, 21 Jul 2023 13:46:43 -0500,
 :ratdec_tc=>"DSBLTYEVALTN",
 :rating_sn=>"401034015",
 :schedule_tc=>"NS",
 :schedule_tn=>"Non-Schedular"}
# many more fields omitted for brevity and PII paranoia
```

Things to note:
- Both the disability name of the primary DC (Tinnitus) and the free-text rating reason (tinnitus with insomnia) are given.
- Two datetimes are given: when the new rating profile was created, and when it was approved/finalized(?) a few hours later. I'm not sure which one would be the effective date if these happen on different days.
- Even though 6260 Tinnitus is the primary DC with a max rating of 10%, the Non-Schedular (`NS`) suggests the reason that this disability is rated higher at 30%.

**Comment by nanotone** [comment]
From sampling a few more rating profiles, it seems that the `schedule_tc` and `schedule_tn` fields only take on a few different values:

- `CS` / `Current Schedule` -- this appears to be the majority
- `NS` / `Non-Schedular` -- infrequent; most Veterans don't have any disabilities of this type
- null / null -- this appears to be the value for disabilities that are Not Service Connected
